### PR TITLE
Fix/do 4418 wrong spelling of the column catenax site id in the table assets as planned

### PR DESCRIPTION
**Pull request by hgo-ds**
fix:DO-4418:rename all properties to catenaXSiteId

- Consolidation on catenaXSiteId
- but the interfaces are tolerant with the spelling

Because the existing possibilities of different spellings of the catenaXSiteId attribute cannot be resolved with certainty, and the uncertainty of a change remains, all relevant places are implemented in the TRX regardless of the spelling.

fix:DO-4418:Local variables should not shadow class fields

Further information provided here: https://github.com/eclipse-tractusx/sldt-semantic-models/issues/470

### CODE REVIEW

File: tx-backend/src/main/java/org/eclipse/tractusx/traceability/assets/infrastructure/base/irs/model/response/semanticdatamodel/Site.java
[the remark this answers is not in the document]
Comment (hgo-ds) [reply tx-backend/src/main/java/org/eclipse/tractusx/traceability/assets/infrastructure/base/irs/model/response/semanticdatamodel/Site.java:1]: Good catch, case insensitivity is not working for some reason, so it is in deed redundant, I'd prefer to drop that line then
`
@JsonFormat(with = JsonFormat.Feature.ACCEPT_CASE_INSENSITIVE_PROPERTIES)
`
and keep the `@JsonAlias`.

File: tx-backend/src/test/java/org/eclipse/tractusx/traceability/integration/common/support/AssetTestData.java
[the remark this answers is not in the document]
Comment (hgo-ds) [reply tx-backend/src/test/java/org/eclipse/tractusx/traceability/integration/common/support/AssetTestData.java:1]: Good point, I'll double check and see what better fits here.

File: tx-backend/src/main/java/org/eclipse/tractusx/traceability/assets/infrastructure/base/irs/model/response/semanticdatamodel/Site.java
[the remark this answers is not in the document]
Comment (hgo-ds) [reply tx-backend/src/main/java/org/eclipse/tractusx/traceability/assets/infrastructure/base/irs/model/response/semanticdatamodel/Site.java:1]: Directly applying JsonFormat.Feature.ACCEPT_CASE_INSENSITIVE_PROPERTIES to fields isn't possible, as it's typically used at the object mapper level or through annotations that are associated with the properties themselves.

**Review by martin-fritz — APPROVED** [review]
Looks perfect to me now.

**Pull request merged**